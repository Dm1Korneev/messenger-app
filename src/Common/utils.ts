
type Filter = Parameters<typeof Array.prototype.filter>[0]
export const distinct: Filter = (value, index, self) => self.indexOf(value) === index;
