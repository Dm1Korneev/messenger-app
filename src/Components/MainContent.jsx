import React from 'react';
import PropTypes from 'prop-types';

// containers
import withStyles from '@material-ui/core/styles/withStyles';
import Grid from '@material-ui/core/Grid';
import MessagesList from 'Containers/MessagesList';
import MessageInput from 'Containers/MessageInput';

// @material-ui

function MainContent(props) {
  const { classes } = props;

  return (
    <main className={classes.main}>
      <Grid item xs={12} sm={11} md={10} lg={9} className={classes.grid}>
        <div className={classes.content}>
          <div className={classes.appBarSpacer} />
          <MessagesList />
          <MessageInput />
        </div>
      </Grid>
    </main>
  );
}

MainContent.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
};

const styles = (theme) => ({
  main: {
    flexGrow: 1,
  },
  grid: {
    margin: '0 auto',
    height: '100%',
  },
  appBarSpacer: { marginTop: theme.spacing.unit, ...theme.mixins.toolbar },
  content: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
});

export default withStyles(styles)(MainContent);
