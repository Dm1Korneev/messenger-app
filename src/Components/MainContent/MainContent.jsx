import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';

import MessageInput from 'Components/MessageInput';
import MessagesList from 'Components/MessagesList';

const useStyles = makeStyles((theme) => ({
  grid: {
    margin: '0 auto',
    height: '100%',
  },
  appBarSpacer: { marginTop: theme.spacing(1), ...theme.mixins.toolbar },
}));

function MainContent() {
  const classes = useStyles();

  return (
    <Box>
      <Grid item xs={12} sm={11} md={10} lg={9} className={classes.grid}>
        <Box display="flex" flexDirection="column" justifyContent="space-between" height="100%">
          <Box className={classes.appBarSpacer} />
          <MessagesList />
          <MessageInput />
        </Box>
      </Grid>
    </Box>
  );
}

export default MainContent;
