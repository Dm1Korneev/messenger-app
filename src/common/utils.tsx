export function distinct(value: any, index: number, self: Array<any>): boolean {
  return self.indexOf(value) === index;
}
