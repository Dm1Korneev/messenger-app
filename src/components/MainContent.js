import React from "react";
import PropTypes from "prop-types";

// containers
import MessagesList from "../containers/MessagesList";
import MessageInput from "../containers/MessageInput";

// @material-ui
import withStyles from "@material-ui/core/styles/withStyles";
import Grid from "@material-ui/core/Grid";

function MainContent(props) {
  const { classes } = props;

  return (
    <main className={classes.main}>
      <Grid item xs={12} sm={11} md={10} lg={9} className={classes.grid}>
        <div className={classes.content}>
          <div className={classes.appBarSpacer} />
          <MessagesList />
          <MessageInput />
        </div>
      </Grid>
    </main>
  );
}

MainContent.propTypes = {
  classes: PropTypes.object.isRequired
};

const styles = theme => ({
  main: {
    flexGrow: 1
  },
  grid: {
    margin: "0 auto",
    height: "100%"
  },
  appBarSpacer: { marginTop: theme.spacing.unit, ...theme.mixins.toolbar },
  content: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between"
  }
});

export default withStyles(styles)(MainContent);
