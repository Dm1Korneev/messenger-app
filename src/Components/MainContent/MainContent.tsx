import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';

import { MessageInput } from 'Components/MessageInput';
import { MessagesList } from 'Components/MessagesList';

const useStyles = makeStyles((theme) => ({
  grid: {
    margin: '0 auto',
    height: '100%',
  },
  appBarSpacer: {
    marginTop: theme.spacing(1),
    ...theme.mixins.toolbar,
  },
}));

type MainContentProps = {
  activeChatId?: string
}

export const MainContent = ({ activeChatId }: MainContentProps) => {
  const classes = useStyles();

  return (
    <Box width="100%">
      <Grid item xs={12} sm={11} md={10} lg={9} className={classes.grid}>
        <Box display="flex" flexDirection="column" justifyContent="space-between" height="100%">
          <Box className={classes.appBarSpacer} />
          <MessagesList activeChatId={activeChatId} />
          {activeChatId && <MessageInput activeChatId={activeChatId} />}
        </Box>
      </Grid>
    </Box>
  );
};
