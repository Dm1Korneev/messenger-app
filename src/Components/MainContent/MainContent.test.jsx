import React from 'react';
import { render } from '@testing-library/react';

import { mockComponent } from 'testing/utils';

import MainContent from './MainContent';

jest.mock('Components/MessagesList', () => mockComponent('MessagesList'));
jest.mock('Components/MessageInput', () => mockComponent('MessageInput'));

jest.mock('@material-ui/core/Grid', () => mockComponent('Grid'));

const props = {};

describe('render MainContent component', () => {
  const setup = () => render(
    <MainContent {...props} />,
  );

  test('snapshot test', async () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});

