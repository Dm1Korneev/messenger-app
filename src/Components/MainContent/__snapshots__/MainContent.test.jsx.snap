// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MainContent component snapshot test 1`] = `
<DocumentFragment>
  <main
    class="MainContent-main-1"
  >
    <div
      data-originalcomponent="Grid"
      test-prop-classname="\\"MainContent-grid-2\\""
      test-prop-item="true"
      test-prop-lg="9"
      test-prop-md="10"
      test-prop-sm="11"
      test-prop-xs="12"
    >
      <div
        class="MainContent-content-4"
      >
        <div
          class="MainContent-appBarSpacer-3"
        />
        <div
          data-originalcomponent="MessagesList"
        />
        <div
          data-originalcomponent="MessageInput"
        />
      </div>
    </div>
  </main>
</DocumentFragment>
`;
